www/student.js

```diff
@@ -86,3 +87,32 @@ endButton.addEventListener('click', function () {
   runButton.disabled = false
   endButton.disabled = true
 })
+
+// Timer functionality
+let timer_running = true
```

Comment:
It might make sense to use `setInterval` and `clearInterval` here so that you don't need to manage the timer state separately.

https://developer.mozilla.org/en-US/docs/Web/API/Window/setInterval